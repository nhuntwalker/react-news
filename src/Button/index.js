import React, { PropTypes } from 'react';
import './index.css';

const Button = ({ onClick, className, children}) =>
  <button onClick={ onClick } className={ className } type="button">
    { children }
  </button>

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

Button.defaultProps = {
  className: '',
}

const Loading = () =>
  <div>
    Loading...
  </div>

const withLoading = (Component) => ({ isLoading, ...rest }) =>
  isLoading ? <Loading /> : <Component { ...rest } />

const ButtonWithLoading = withLoading(Button);

export default Button;

export {
    ButtonWithLoading,
};