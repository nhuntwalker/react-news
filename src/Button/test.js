import React from 'react';
import ReactDOM from 'react-dom';
import Button from './';
import renderer from 'react-test-renderer';

describe('Button', () => {
    const onClick = () => true;
    it('renders', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Button onClick={ onClick }>Give Me More</Button>, div);
    });

    test('snapshots', () => {
        const component = renderer.create(<Button onClick={ onClick }>Give Me More</Button>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});