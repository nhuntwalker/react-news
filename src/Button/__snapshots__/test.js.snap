exports[`Button snapshots 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button">
  Give Me More
</button>
`;
